notebooks/IMDB to Hotel Sentiment.ipynb

```diff
@@ -0,0 +1,725 @@
+   "id": "139763f8",
+   "metadata": {},
+   "source": [
+    "For our generator, we use OpenAI's GPT-3 model. For this, we need to read the access key in from a file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a6293838",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "with open(os.path.expanduser('~/.openai_api_key'), 'r') as file:\n",
+    "    OPENAI_API_KEY = file.read().replace('\\n', '')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "70f74aec",
+   "metadata": {},
+   "source": [
+    "First, we create the generator object which AdaTest will use to suggest more tests which are similar to the ones we provide:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6e310578",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "generator = adatest.generators.OpenAI('curie', api_key=OPENAI_API_KEY)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "56095496",
+   "metadata": {},
+   "source": [
+    "Now we create the test tree. We will load a set of tests which we have alreadys started work on, to make the process faster:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1bc4e9ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tests = adatest.TestTree(\"imdb_hotel_conversion.csv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "58006371",
+   "metadata": {},
+   "source": [
+    "And fire up the AdaTest interface:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "056dbba0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tests.adapt(original_pipeline, generator, auto_save=True, recompute_scores=True)"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "id": "b61cb927",
+   "metadata": {},
+   "source": [
+    "With a set of samples composed, we need to use them to finetune the model. To begin this process, load the CSV file we've created into a DataFrame and drop the portions we don't need:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "32a790c1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "def load_adatest_data(csv_file: str):\n",
+    "    tmp = pd.read_csv(csv_file)\n",
+    "    \n",
+    "    # Drop topic marker rows\n",
+    "    tmp2 = tmp[tmp['label'] != 'topic_marker']\n",
+    "    # Drop suggestion rows\n",
+    "    tmp3 = tmp2[tmp2['topic'] != 'suggestion']\n",
+    "    \n",
+    "    # Remove columns we don't need\n",
+    "    tmp4 = tmp3.drop(labels=['labeler', 'description', 'author', 'Unnamed: 0'], axis=1)\n",
+    "    \n",
+    "    # Rename columns\n",
+    "    tmp5 = tmp4.rename(mapper={'input': 'sentence', 'label': 'model_is_correct'}, axis=1)\n",
+    "    \n",
+    "    # Don't need to track original rows\n",
+    "    tmp6 = tmp5.reset_index(drop=True)\n",
+    "    \n",
+    "    return tmp6\n",
```

Comment:
I don't think we should end up with an 'Unnamed: 0' in the columns when we read CSV -- maybe we need to set header or index to false on the CSV load?

Reply:
It's not that. It's the CSV itself:
```
,topic,input,output,label,labeler,description,author,model score
00f2dfd40977438f9942793768959e8a,suggestion,,,topic_marker,imputed,,,
02076434c5fc4a389f127e542c479c64,/Rooms,Sheets didn't seem clean,neg,pass,anonymous,,anonymous,-0.9646334648132324
```
There is no name for the column of GUIDs. That's what's being dropped.

I'll have this down as a "Won't fix" for this notebook. Perhaps AdaTest should put a name on that column, though.